packages/jest-haste-map/src/index.js

```diff
@@ -505,36 +509,23 @@ class HasteMap extends EventEmitter {
   /**
    * Creates workers or parses files and extracts metadata in-process.
    */
-  _getWorker(
-    options: ?{forceInBand: boolean},
-  ): (message: WorkerMessage) => Promise<WorkerMetadata> {
-    if (!this._workerPromise) {
-      let workerFn;
+  _getWorker(options: ?{forceInBand: boolean}): WorkerInterface {
+    if (!this._worker) {
+      const workerOptions = {
```

Comment:
seems like you can move these options into the `else` block.

Reply:
Actually, can you just inline them into the worker instantiation? No need for vars!